app/src/main/java/com/dragic/gamehunter/view/uicomponents/TopBar.kt

```diff
@@ -0,0 +1,98 @@
+package com.dragic.gamehunter.view.uicomponents
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import com.dragic.gamehunter.R
+import com.dragic.gamehunter.view.navigation.Details
+import com.dragic.gamehunter.view.navigation.Home
+import com.dragic.gamehunter.view.theme.Typography
+
+@Composable
+fun TopBar(
```

Comment:
Instead of having one top bar that handles every screen, create a separate top bar for each screen. That way you don't have to have these if-checks. It is also more flexible and simple.